community_usecase/Profile_generation/run_profile_generation.py

```diff
@@ -0,0 +1,291 @@
+    logger.info("============ Analysis Complete ============\n")
+
+    return final_summary
+
+# ------------------ Pydantic Schemas ------------------
+
+class PersonalInformation(BaseModel):
+    name: str
+    positions: list[str]
+    contact_information: str | None = None
+    social_media: list[str] | None = None
+    research_keywords: list[str]
+
+
+class Biography(BaseModel):
+    career_history: str
+    research_focus: str
+
+
+class ResearchInterests(BaseModel):
+    areas: list[str]
+
+
+class AwardsAndDistinctions(BaseModel):
+    honors: list[str]
+
+
+class Education(BaseModel):
+    degrees: list[str]
+
+
+class ScholarProfile(BaseModel):
+    personal_information: PersonalInformation
+    biography: Biography
+    research_interests: ResearchInterests
+    awards_and_distinctions: AwardsAndDistinctions
+    education: Education
+
+
+def generate_html_profile(input_text: str,
+                          output_file: str = "profile.html") -> None:
+    """Pipeline: extract structured profile from a free‑form biography and
+    turn it into a polished HTML page.
+
+    Parameters
+    ----------
+    input_text : str
+        Raw biography or resume text.
+    output_file : str, optional
+        Path where the final HTML file will be saved, by default
+        "profile.html".
+    """
+    extraction_prompt = (
+        "You are a scholarly‑profile extraction assistant. "
+        "Return ONLY JSON that matches the provided schema."
+    )
+
+    extractor = ChatAgent(system_message=extraction_prompt)
+    extraction_response = extractor.step(
+        input_message=input_text,
+        response_format=ScholarProfile,
+    )
+    profile: ScholarProfile = extraction_response.msgs[0].parsed
+
+    html_system_prompt = "You are an expert HTML profile page generator."
+    html_agent = ChatAgent(system_message=html_system_prompt)
+
+    html_user_prompt = f"""
+    Using the following JSON, build a complete HTML5 profile page.
+
+    • Sections: Personal Information, Biography, Research Interests, Awards 
+    and Distinctions, Education.
+    Generate a professional and natural-looking personal profile HTML page 
+    based 
+    on the provided information, following these detailed requirements:
+    Using standard HTML5 + simple inline CSS, beautiful but not complicated.
+    Layout should be clearly organized using <h1> for the main title, <h2> for 
+    section headings, and <p> for paragraph text.
+    Use proper <a href> hyperlinks for any external links (e.g., LinkedIn, 
+    ResearchGate, Google Scholar).
+    Write in a natural, flowing narrative style suitable for introducing the 
+    person to a broad audience.
+
+    Please write ALL the input content to the webpage.
+    JSON:
+    {profile.model_dump_json(indent=2)}
+    """
+
+    html_response = html_agent.step(html_user_prompt)
+    html_code = html_response.msgs[0].content
+    html_code_block = re.search(r"```html(.*?)```", html_code, re.DOTALL)
```

Comment:
Can we not use agent for generating HTML? Just use a pre-defined template for this